api/plot/utils.py

```diff
@@ -155,14 +162,17 @@ def image(
         image: array
         x: x array
         y: y array
-        domain: array
         title: title of plot
         Returns
         -------
         response: Response
             Response from push_data POST request
         """
-        im = ImageData(key="", values=np.asanyarray(image), domain=domain, **attribs)
+        values = np.asanyarray(image)
+        if "domain" in attribs:
+            im = HeatmapData(key="", values=values, **attribs)
+        elif (len(values.shape) == 3 and values.shape[2] == 3):
```

Comment:
Use `values.ndim == 3`. Also check values is 2D in heatmap case, and if not either case then raise error